### ISSUE TITLE
Reverse populate?

### ISSUE THREAD

**Issue by jbonnett92** [moleculerjs/moleculer-db#55]
Hi,
We can populate by using a value from a particular field, how would we populate if we didn't have that field? E.g. I have two databases conversations and messages where messages references a conversation although if I wanted to get the messages for a conversation that is a different story.
Is there a way around this?

Thanks,
Jamie

**Comment by jbonnett92** [comment]
Thank you!
How would I go about populating the virtual field? I did try this:
```
has: {
	params: {
		conversation: { type: "string" }
	},
	handler(ctx) {
		let params = {
			sort: ["createdAt"],
			populate: ["author"],
			query: { conversation: ctx.params.conversation }
		};

		return this.adapter.find(params);
	}
},
```

**Comment by jbonnett92** [comment]
Ahh I forgot to transform the result `return this.adapter.find(params).then(doc => this.transformDocuments(ctx, params, doc));`

**Comment by icebob** [comment]
it means, the response comes from the cache. If you got the same result for different request means, you should fine-tune the cache key fields. More info: http://moleculer.services/docs/0.13/caching.html#Cache-keys

**Comment by jbonnett92** [comment]
I don't have caching fields added. So it shouldn't cache, I purposely removed them for development.

**Comment by icebob** [comment]
But the yellow `*` [means](https://github.com/moleculerjs/moleculer-metrics/blob/master/packages/moleculer-console-tracer/src/index.js#L137) the response came from cache

**Comment by jbonnett92** [comment]
It turns out I had this in my config which was the problem:
```
cacher: {
	type: "memory",
	options: {
		maxParamsLength: 100
	}
},
```

The weird thing is that it wasn't caching everything and as I said I removed the cache fields so is that strange behaviour?

Just in case you're wondering I'm working on a chat application :) [video here](https://i.imgur.com/gl0hzZx.mp4).

**Comment by icebob** [comment]
Yes, this config enable caching, and moleculer-db base query actions have [enabled caching by default](https://github.com/moleculerjs/moleculer-db/blob/master/packages/moleculer-db/src/index.js#L77). But if you extend it with custom actions they won't be caching by default, just if you set it.

Nice project. Is it a private or public project? 
I started a [kanban project](https://github.com/icebob/kantab) with the same tech stack :). But I stucked in the authentications :(
